qemu/tests/nic_opt.py

```diff
@@ -0,0 +1,149 @@
+                else:
+                    test.error("Can not start netperf client.")
+                num += 1
+                # here when set a run flag, when other case call this case as a
+                # subprocess backgroundly,can set this run flag to False to stop
+                # the stress test.
+                env["netperf_run"] = True
+                netperf_test_duration = stop_time - time.time()
+                utils_misc.wait_for(
+                    lambda: not n_client.is_netperf_running(),
+                    netperf_test_duration, 0, 5,
+                    "Wait netperf test finish %ss" % netperf_test_duration)
+                time.sleep(5)
+        finally:
+            n_server.stop()
+            n_server.package.env_cleanup(True)
+            n_client.package.env_cleanup(True)
+            env["netperf_run"] = False
+
+    def test_ping():
+        """
+        Ping test for nic option.
+        """
+        package_sizes = params.objects("ping_sizes")
+        if params["os_type"] == "windows":
+            ifname = utils_net.get_windows_nic_attribute(
+                session=session,
+                key="netenabled",
+                value=True,
+                target="netconnectionID")
+        else:
+            ifname = utils_net.get_linux_ifname(session,
+                                                vm.get_mac_address())
+        dest = utils_net.get_host_ip_address(params)
+
+        for size in package_sizes:
+            error_context.context(
+                "Test ping from '%s' to host '%s' on guest '%s'"
+                " with package size %s. " %
+                (ifname, dest, vm.name, size), logging.info)
+            status, output = utils_net.ping(dest=dest, count=10,
+                                            packetsize=size,
+                                            session=session,
+                                            timeout=30)
+            if status:
+                test.fail("%s ping %s unexpected, output %s" % (vm.name, dest,
+                                                                output))
+            if params["os_type"] == "windows":
+                # windows guest get loss=0 when ping failed sometime,
+                # need further check
+                loss = utils_test.get_loss_ratio(output)
+                if not loss and "TTL" in output:
+                    pass
+                else:
+                    test.fail("Guest ping test hit unexpected loss, "
+                              "error_info: %s" % output)
+
+    check_type = params.get("check_type")
+    smp_value = params.get_numeric("smp") or params.get_numeric("vcpu_maxcpus")
+    if cpu.online_cpus_count() < 2 * smp_value:
+        test.cancel("The number of smp counts in this host is not big enough")
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_serial_login()
+    if params["os_type"] == "windows":
```

Comment:
close session in finally code:

```
try:
  ....
finally:
       session.close()
```

Reply:
@xutian please help to review it again,thanks.